From f5fe0ec294c62a4b7c58accae2262fb4b1c2af30 Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Thu, 30 Mar 2017 13:43:05 +0800
Subject: [PATCH] [MMFMWK-7338]subparse: fix critical log print out when set
 rate <0 with external subtitle

When playback with external subtitle, subparse will force its own start
and stop position when seek. When rate < 0, critial log will print out
assertion 'stop != -1', so need avoid it when rate < 0.

Upstream-Status: Pending
https://bugzilla.gnome.org/show_bug.cgi?id=771648
---
 gst/subparse/gstsubparse.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gst/subparse/gstsubparse.c b/gst/subparse/gstsubparse.c
index 0fc00dd..565b5bb 100755
--- a/gst/subparse/gstsubparse.c
+++ b/gst/subparse/gstsubparse.c
@@ -279,6 +279,11 @@ gst_sub_parse_src_event (GstPad * pad, GstObject * parent, GstEvent * event)
 
       /* Convert that seek to a seeking in bytes at position 0,
          FIXME: could use an index */
+      if (rate < 0) {
+        rate = 1.0;
+        GST_WARNING_OBJECT (self, "Only can push positive rate upstream");
+      }
+
       ret = gst_pad_push_event (self->sinkpad,
           gst_event_new_seek (rate, GST_FORMAT_BYTES, flags,
               GST_SEEK_TYPE_SET, 0, GST_SEEK_TYPE_NONE, 0));
-- 
1.9.1

